cstar/base/additional_code.py

```diff
@@ -86,39 +94,31 @@ def __init__(
 
         """
 
-        # TODO:  Type check here
         self.base_model: BaseModel = base_model
         self.source: DataSource = DataSource(location)
+        self.subdir: str = subdir
         self.checkout_target: Optional[str] = checkout_target
         self.source_mods: Optional[List[str]] = source_mods
         self.namelists: Optional[List[str]] = namelists
-        self.exists_locally: Optional[bool] = None
-        self.local_path: Optional[Path] = None
+        self.working_path: Optional[Path] = None
 
         # If there are namelists, make a parallel attribute to keep track of the ones we are editing
         # AdditionalCode.get() determines which namelists are editable templates and updates this list
         if self.namelists:
             self.modified_namelists: list = []
 
-        if self.source.location_type == "path":
-            self.exists_locally = True
-            self.local_path = Path(self.source.location).resolve()
-
     def __str__(self) -> str:
-        base_str = (
-            "AdditionalCode"  # associated with {self.base_model.name} base model"
-        )
-        base_str += "\n---------------------"
+        base_str = self.__class__.__name__ + "\n"
+        base_str += "-" * (len(base_str) - 1)
         base_str += f"\nBase model: {self.base_model.name}"
         base_str += f"\nLocation: {self.source.location}"
-        if self.exists_locally is not None:
-            base_str += f"\n Exists locally: {self.exists_locally}"
-        if self.local_path is not None:
-            base_str += f"\n Local path: {self.local_path}"
+        base_str += f"\nsubdirectory: {self.subdir}"
+        base_str += f"\nWorking path: {self.working_path}"
+        base_str += f"\nExists locally: {self.exists_locally}"
+        if not self.exists_locally:
+            base_str += " (get with AdditionalCode.get())"
```

Comment:
This might be a little more readable
```suggestion
            base_str.append(" (get with AdditionalCode.get())")
```